engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -364,6 +404,57 @@ protected static IntConsumer getChunkNullSetter(final WritableChunk<? extends Va
         }
     }
 
+    /**
+     * This assigns the value at {@code index} in {@code valueChunk} to the {@code inputColumnSource}. This is not
+     * particularly efficient but is only be called once per bucket.
+     */
+    private static void assignKeyValue(
```

Comment:
Does this mean we're reading key values many times, redundantly, in order to capture them once per bucket for our scalar inputs?

Reply:
Hmmm... good point. Suggestions?